substrate/frame/assets/src/lib.rs

```diff
@@ -456,6 +464,33 @@ pub mod pallet {
 		ValueQuery,
 	>;
 
+	#[pallet::storage]
+	/// Merklized distribution of an asset.
+	pub(super) type CountForMerklizedDistribution<T: Config<I>, I: 'static = ()> =
```

Comment:
Should this not be a counted map?

Reply:
@kianenigma I use the index as a uuid for the distribution. It is possible to delete the item from the map at the end of a distribution to clean everything up.

So in this case, the counted map would decrease the count, when i want a strictly monotonically increasing number.